src/mqtt_client/mod.rs

```diff
@@ -339,58 +409,82 @@ where
     {
         let settings = &mut self.settings;
         let mqtt = &mut self.mqtt;
-        let prefix = self.settings_prefix.as_str();
-
-        let mut response_topic: String<MAX_TOPIC_LENGTH> = String::from(self.prefix.as_str());
-        response_topic.push_str("/log").unwrap();
-        let default_response_topic = response_topic.as_str();
+        let prefix = self.prefix.as_str();
+        let listing_state = &mut self.listing_state;
 
         let mut updated = false;
         match mqtt.poll(|client, topic, message, properties| {
-            let path = match topic.strip_prefix(prefix) {
-                // For paths, we do not want to include the leading slash.
-                Some(path) => {
-                    if !path.is_empty() {
-                        &path[1..]
-                    } else {
-                        path
-                    }
+            let Some(path) = topic.strip_prefix(prefix) else {
+                info!("Unexpected MQTT topic: {}", topic);
+                return;
+            };
+
+            let Ok(command) = Command::from_str(path) else {
+                info!("Unknown Miniconf command: {path}");
+                return;
+            };
+
+            let mut data = [0u8; MESSAGE_SIZE];
+            let message = match command {
+                Command::Query => {
```

Comment:
Can we stick to `Get` as in the miniconf trait?

Reply:
Sure. I was hopping back and forth between the naming, but `get` is definitely simpler. In reality, I imagine tooling will abstract all this so the user won't even notice it.